test/integration/targets/pip/tasks/vcs_packages.yml

```
@@ -0,0 +1,185 @@
+      content: |
+        def return_one():
+            return 1
+
+  - name: Create the subdirectory for the nested package
+    file:
+      path: "{{ repo_path }}/dummy-root/dummy-subdirectory"
+      state: directory
+      mode: '0755'
+
+  - name: Create nested package's pyproject.toml
+    copy:
+      dest: "{{ repo_path }}/dummy-root/dummy-subdirectory/pyproject.toml"
+      content: |
+        [build-system]
+        requires = ["setuptools>=61.0"]
+        build-backend = "setuptools.build_meta"
+
+        [project]
+        name = "nested-package"
+        version = "0.0.1"
+        dependencies = []
+
+  - name: Create nested package's source directory
+    file:
+      path: "{{ repo_path }}/dummy-root/dummy-subdirectory/src"
+      state: directory
+      mode: '0755'
+
+  - name: Create nested package's source file
+    copy:
+      dest: "{{ repo_path }}/dummy-root/dummy-subdirectory/src/nested_package.py"
+      content: |
+        def return_two():
+            return 2
+
+  - name: Create the directory for the egg-formatted package
+    file:
+      path: "{{ repo_path }}/dummy-root/dummy-egg-package"
+      state: directory
+      mode: '0755'
+
+  - name: Create egg-formatted package's setup.py
+    copy:
+      dest: "{{ repo_path }}/dummy-root/dummy-egg-package/setup.py"
+      content: |
+        from setuptools import setup, find_packages
+        setup(
+            name='egg-package',
+            version='0.0.1',
+            packages=find_packages(),
+        )
+
+  - name: Create the source directory for the egg package
+    file:
+      path: "{{ repo_path }}/dummy-root/dummy-egg-package/egg_package"
+      state: directory
+      mode: '0755'
+
+  - name: Create egg-formatted package's source file
+    copy:
+      dest: "{{ repo_path }}/dummy-root/dummy-egg-package/egg_package/utility.py"
+      content: |
+        def return_three():
+            return 3
+
+  - name: Add __init__.py files
+    file:
+      path: "{{ item }}"
+      state: touch
+    with_items:
+      - "{{ repo_path }}/dummy-root/__init__.py"
+      - "{{ repo_path }}/dummy-root/dummy-subdirectory/__init__.py"
+      - "{{ repo_path }}/dummy-root/dummy-egg-package/__init__.py"
+      - "{{ repo_path }}/dummy-root/dummy-egg-package/egg_package/__init__.py"
+
+  - name: Create git repository
+    command:
+      cmd: "{{ item }}"
+    args:
+      chdir: "{{ repo_path }}"
+    environment:
+      GIT_CONFIG_NOSYSTEM: 1
+    loop:
+      - git init .
+      - git add .
+      - git -c user.name="Ansible Test" -c user.email="ansible@ansible.ansible" commit -m "First commit"
+
+  - name: Ensure fresh venv
+    file:
+      path: "{{ remote_tmp_dir }}/gitvenv"
+      state: absent
+
+  - name: Create fresh venv with no packages
+    pip:
+      name: []
+      virtualenv: "{{ remote_tmp_dir }}/gitvenv"
+
+  - name: Check installation of packages via git
+    pip:
+      name: "git+file://{{ repo_path }}"
```

Use the bare repo instead of a checkout dir:
```suggestion
      name: "git+file://{{ repo_path }}/.git"
```